include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -74,6 +77,81 @@
 DEAL_II_NAMESPACE_OPEN
 
 
+namespace
+{
+  template <int dim, int fe_degree, int components, typename number>
+  class MassOperator : public Subscriptor
```

Comment:
you can also derive it from `Base` and add diagonal to have a better preconditioner. Not that it will matter much in this context, but still.